app/moves/middleware.js

```diff
@@ -39,6 +39,16 @@ module.exports = {
     next()
   },
   setFromLocation: (req, res, next, locationId) => {
+    const userLocations = get(req.session, 'user.locations')
+    const location = find(userLocations, { id: locationId })
+
+    if (!location) {
+      const error = new Error('Location not found')
+      error.statusCode = 404
```

Comment:
Unfortunately we need to set the `statusCode` otherwise it will default to `500` in the error middleware that gets called as soon as you pass an error to `next()`.